src/joint_trajectory_action/joint_trajectory_action.py

```diff
@@ -354,6 +354,19 @@ def _on_trajectory_action(self, goal):
         control_rate = rospy.Rate(self._control_rate)
 
         dimensions_dict = self._determine_dimensions(trajectory_points)
+
+        if num_points == 1:
+            # Add current position as trajectory point
+            first_trajectory_point = JointTrajectoryPoint()
+            first_trajectory_point.positions = self._get_current_position(joint_names)
+            if dimensions_dict['velocities']:
```

Comment:
in the next stanza, you force the final trajectory point velocities and accelerations to 0 ... yet you check for them here? Should there be something else going on instead? like assigning the velocities over to the first point or using a middle point?